_ml-commons-plugin/semantic-search.md

```diff
@@ -0,0 +1,1039 @@
+  "status": "CREATED"
+}
+```
+
+You'll use this ID to register the chosen model to the model group.
+
+<details closed markdown="block">
+  <summary>
+    Test it
+  </summary>
+  {: .text-delta}
+
+Search for the newly created model group by providing its model group ID in the request:
+
+```json
+POST /_plugins/_ml/model_groups/_search
+{
+  "query": {
+    "match": {
+      "_id": "Z1eQf4oB5Vm0Tdw8EIP2"
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+The response contains the model group:
+
+```json
+{
+  "took": 0,
+  "timed_out": false,
+  "_shards": {
+    "total": 1,
+    "successful": 1,
+    "skipped": 0,
+    "failed": 0
+  },
+  "hits": {
+    "total": {
+      "value": 1,
+      "relation": "eq"
+    },
+    "max_score": 1,
+    "hits": [
+      {
+        "_index": ".plugins-ml-model-group",
+        "_id": "Z1eQf4oB5Vm0Tdw8EIP2",
+        "_version": 1,
+        "_seq_no": 14,
+        "_primary_term": 2,
+        "_score": 1,
+        "_source": {
+          "created_time": 1694357262582,
+          "access": "public",
+          "latest_version": 0,
+          "last_updated_time": 1694357262582,
+          "name": "NLP_model_group",
+          "description": "A model group for NLP models"
+        }
+      }
+    ]
+  }
+}
+```
+</details>
+
+
+### Step 1(c): Register the model to the model group
+
+The DistilBERT model you'll use To register the model to the model group, provide the model group ID in the register request:
+
+```json
+POST /_plugins/_ml/models/_register
+{
+  "name": "huggingface/sentence-transformers/msmarco-distilbert-base-tas-b",
+  "version": "1.0.1",
+  "model_group_id": "Z1eQf4oB5Vm0Tdw8EIP2",
+  "model_format": "TORCH_SCRIPT"
+}
+```
+{% include copy-curl.html %}
+
+Registering a model is an asynchronous task. OpenSearch sends back a task ID for this task:
+
+```json
+{
+  "task_id": "aFeif4oB5Vm0Tdw8yoN7",
+  "status": "CREATED"
+}
+```
+
+OpenSearch downloads the config file for the model and the model contents from the URL. Because the model is larger than 10 MB in size, OpenSearch splits it into chunks of up to 10 MB and saves those chunks in the model index. You can check the status of the task using the Tasks API:
+
+```json
+GET /_plugins/_ml/tasks/aFeif4oB5Vm0Tdw8yoN7
+```
+{% include copy-curl.html %}
+
+Once the task is complete, the task status will be `COMPLETED` and the Tasks API response will contain a model ID for the registered model:
+
+```json
+{
+  "model_id": "aVeif4oB5Vm0Tdw8zYO2",
+  "task_type": "REGISTER_MODEL",
+  "function_name": "TEXT_EMBEDDING",
+  "state": "COMPLETED",
+  "worker_node": [
+    "4p6FVOmJRtu3wehDD74hzQ"
+  ],
+  "create_time": 1694358489722,
+  "last_update_time": 1694358499139,
+  "is_async": true
+}
+```
+
+You'll need the model ID in order to use this model for several following steps.
+
+<details closed markdown="block">
+  <summary>
+    Test it
+  </summary>
+  {: .text-delta}
+
+Search for the newly created model by providing its ID in the request:
```

Comment:
I think we can just use get model API to get the newly created model by model id